hugegraph-common/src/main/java/org/apache/hugegraph/rest/AbstractRestClient.java

```diff
@@ -17,188 +17,245 @@
-                              String trustStorePassword) {
-        this(url, new ConfigBuilder().configTimeout(timeout)
-                                     .configToken(token)
-                                     .configPool(maxTotal, maxPerRoute)
-                                     .configSSL(trustStoreFile,
-                                                trustStorePassword)
-                                     .build());
-    }
-
-    public AbstractRestClient(String url, ClientConfig config) {
-        configConnectionManager(url, config);
-
-        this.client = JerseyClientBuilder.newClient(config);
-        this.client.register(GZipEncoder.class);
-        this.target = this.client.target(url);
-        this.pool = (PoolingHttpClientConnectionManager) config.getProperty(
-                    ApacheClientProperties.CONNECTION_MANAGER);
-        if (this.pool != null) {
-            this.cleanExecutor = ExecutorUtil.newScheduledThreadPool(
-                                              "conn-clean-worker-%d");
-            Number idleTimeProp = (Number) config.getProperty("idleTime");
-            final long idleTime = idleTimeProp == null ?
-                                  IDLE_TIME : idleTimeProp.longValue();
-            final long checkPeriod = idleTime / 2L;
-            this.cleanExecutor.scheduleWithFixedDelay(() -> {
-                PoolStats stats = this.pool.getTotalStats();
-                int using = stats.getLeased() + stats.getPending();
-                if (using > 0) {
-                    // Do clean only when all connections are idle
-                    return;
-                }
-                // Release connections when all clients are inactive
-                this.pool.closeIdleConnections(idleTime, TimeUnit.MILLISECONDS);
-                this.pool.closeExpiredConnections();
-            }, checkPeriod, checkPeriod, TimeUnit.MILLISECONDS);
+    private static String parseContentType(RestHeaders headers) {
+        if (headers != null) {
+            String contentType = headers.get(HttpHeadersConstant.CONTENT_TYPE);
+            if (contentType != null) {
+                return contentType;
+            }
         }
+        return HttpHeadersConstant.APPLICATION_JSON;
     }
 
-    protected abstract void checkStatus(Response response,
-                                        Response.Status... statuses);
+    private OkHttpClient buildOkHttpClient(OkHttpConfig okHttpConfig) {
+        OkHttpClient.Builder builder = new OkHttpClient.Builder();
+
+        if (okHttpConfig.getTimeout() != null) {
+            builder.connectTimeout(okHttpConfig.getTimeout(), TimeUnit.MILLISECONDS)
+                   .readTimeout(okHttpConfig.getTimeout(), TimeUnit.MILLISECONDS);
+        }
+
+        if (okHttpConfig.getIdleTime() != null) {
+            ConnectionPool connectionPool =
+                    new ConnectionPool(5, okHttpConfig.getIdleTime(), TimeUnit.MILLISECONDS);
```

Comment:
please note the 5 in `ConnectionPool(5, ...)`, can we pass it from OkHttpConfig?
can we also add a test case for it?